File: DevExpress-Demo/VB/ChartsDemo.Wpf/Modules/TabbedModuleViewModel.vb
Imports System.Windows.Controls
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Core

Namespace ChartsDemo

    Public Class TabbedModuleViewModel

        Public Shared Function Create() As TabbedModuleViewModel
            Return ViewModelSource.Create(Function() New TabbedModuleViewModel())
        End Function

        Public Overridable Property SelectedTab As DXTabItem

        Protected Sub New()
        End Sub
    End Class

    Public Class NotCachedTabbedModuleViewModel

        Public Shared Function Create() As NotCachedTabbedModuleViewModel
            Return ViewModelSource.Create(Function() New NotCachedTabbedModuleViewModel())
        End Function

        Public Overridable Property SelectedTab As DXTabItem

        Public Overridable Property Options As StackPanel
    End Class
End Namespace

File: DevExpress-Demo/CS/ChartsDemo.Wpf/Modules/TabbedModuleViewModel.cs
using System.Windows.Controls;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Core;

namespace ChartsDemo {
    public class TabbedModuleViewModel {
        public static TabbedModuleViewModel Create() {
            return ViewModelSource.Create(() => new TabbedModuleViewModel());
        }

        public virtual DXTabItem SelectedTab { get; set; }

        protected TabbedModuleViewModel() { }
    }


    public class NotCachedTabbedModuleViewModel {
        public static NotCachedTabbedModuleViewModel Create() {
            return ViewModelSource.Create(() => new NotCachedTabbedModuleViewModel());
        }

        public virtual DXTabItem SelectedTab { get; set; }
        public virtual StackPanel Options { get; set; }
    }
}
